### PR TITLE
added support of inputs in navigateScreenAPI

### LINKED ISSUE #309: Support for passing inputs to navigateScreen from ensemble object
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by sagardspeed2**
## Issue ticket number and link

Closes #
closes #309 

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests
- [x] I have added a changeset `pnpm changeset add`
- [x] I have added example usage in the kitchen sink app

### CODE REVIEW

File: packages/runtime/src/runtime/navigateApi.tsx [hunk 1]
```diff
@@ -22,5 +28,18 @@ export const navigateApi = (
     return;
   }
 
-  navigate(`/${matchingScreen.name.toLowerCase()}`);
+  // set additional inputs
+  const inputs = !isString(targetScreen) && data ? cloneDeep(data) : {};
```
Comment (sibteali786) [inline packages/runtime/src/runtime/navigateApi.tsx:32]: @sagardspeed2 i am not sure what syntax you are gonna support but i was looking if we can use 
```js
ensemble.navigateScreen("{name: 'home', inputs: {id: 4}}");
```
@evshi thoughts ?

Comment (sagardspeed2) [reply packages/runtime/src/runtime/navigateApi.tsx]: @sibteali786 currently this syntax will support the inputs.
```
ensemble.navigateScreen('screenName', payload)
```

@sibteali786 @evshi your thoughts on this ?

Comment (evshi) [reply packages/runtime/src/runtime/navigateApi.tsx]: It should follow the existing convention set by flutter side. Please check existing schema or docs if you're unsure https://docs.ensembleui.com/#/actions/8-navigateScreen?id=propertiesoptions

```yaml
      - Button:
          label: Navigate and pass inputs using code
          onTap:
            executeCode:
              body: |
                //@code
                ensemble.navigateScreen({
                  "name": "Action: navigateScreen inputs demo",
                  "inputs":  { 
                    "message": messageInput.value
                  }
                });

```

File: packages/runtime/src/runtime/navigateApi.tsx [hunk 2]
```diff
@@ -1,26 +1,41 @@
...
+  findExpressions(inputs, [], expressionMap);
+  expressionMap.forEach(([path, value]) => {
+    const result = evaluate(screenContext, value, {
+      ensemble: { storage },
+      ...customScope,
+    });
+    set(inputs, path, result);
+  });
+
+  navigate(`/${matchingScreen.name.toLowerCase()}`, { state: inputs });
```
Comment (evshi) [inline packages/runtime/src/runtime/navigateApi.tsx:40]: You don't need to do the evaluation, just update the function signature and pass the state to `navigate`. The rest is taken care of by the `evaluate` fn

File: packages/runtime/src/runtime/hooks/useEnsembleAction.tsx [hunk 3]
```diff
@@ -111,8 +112,15 @@ export const useExecuteCode: EnsembleActionHook<
...
-                navigateScreen: (targetScreen: NavigateScreenAction): void =>
-                  navigateApi(targetScreen, screen, navigate),
+                navigateScreen: (targetScreen: string, data: unknown): void =>
+                  navigateApi(
+                    targetScreen,
+                    data,
+                    screen,
+                    storage,
+                    navigate,
+                    customScope,
```
Comment (evshi) [inline packages/runtime/src/runtime/hooks/useEnsembleAction.tsx:122]: This scope should actually be passed in to the evaluate fn. Your navigate fn just serves as a type of callback, it doesn't do any evaluation itself

File: packages/runtime/src/runtime/navigateApi.tsx (hunk 1 above)
Comment (sagardspeed2) [reply packages/runtime/src/runtime/navigateApi.tsx]: okay, will update this as per the flutter schema

**Review by evshi — APPROVED** [review]

**Pull request merged**